cardano-crypto-praos/src/Cardano/Crypto/VRF/Praos.hs

```diff
@@ -418,6 +423,26 @@ prove sk msg =
             0 -> return $ Just $! proof
             _ -> return Nothing
 
+-- | Construct a BatchCompat vkey from praos, non-batchcompat
+vkToBatchCompat :: VerKeyVRF PraosVRF -> VerKeyVRF BC.PraosBatchCompatVRF
+vkToBatchCompat praosVk =
+  case rawDeserialiseVerKeyVRF (rawSerialiseVerKeyVRF praosVk) of
+    Just vk -> vk
+    Nothing -> error "All valid Praos VKs are valid BatchCompat VKs."
```

Comment:
Wouldn't it make more sense to word this the other way around? That is, the error condition is that the given VK is *not* a valid batch-compatible VK for some reason. I.o.w., the error message should state the error condition, not the unmet assertion. (This is unlike property check names, where we do name property check functions after the desired property.)